src/Hamiltonians/ReducedDensityMatrix.jl

```diff
@@ -0,0 +1,150 @@
+"""
+    SingleParticleReducedDensityMatrix(i, j) <: AbstractHamiltonian
+
+Represent a {i,j} element of the single-particle reduced density matrix:
+
+```math
+\\hat{ρ}^{(1)}_{i,j} = \\langle \\psi | \\hat a^†_{i} \\hat a_{j} | \\psi \\rangle
+```
+
+where ``i`` and ``j`` are the `mode` and ``| \\psi \\rangle`` is the `state-ket` of the given Hamiltonian.
+
+# See also
+
+* [`single_particle_density`](@ref)
+* [`SingleParticleDensity`](@ref)
+* [`TwoParticleReducedDensityMatrix`](@ref)
+"""
+struct SingleParticleReducedDensityMatrix <: AbstractHamiltonian{Float64}
+    I::Int
+    J::Int
```

Comment:
I'd prefer to use lowercase names for the indices. Uppercase is usually reserved for type names and parameters and sometimes matrices.